enhancements/network/on-prem-mutable-vips.md

```diff
@@ -0,0 +1,199 @@
+
+### baremetal-runtimecfg
+
+Keepalived configuration is rendered based on the command-line parameters provided to the baremetal-runtimecfg. Those come from the Pod definition that is managed by the MCO. Because of that it is valid to say that baremetal-runtimecfg is oblivious to whether `PlatformSpec` or `PlatformStatus` stores the desired config.
+
+### Test Plan
+
+We will need to add a few tests that will perform the operation of adding a pair of VIPs to an already existing dual-stack clusters. Once done, we can reuse the already existing steps that test dual-stack clusters with dual-stack VIPs.
+
+### Graduation Criteria
+
+We do not anticipate needing a graduation process for this feature. The internal loadbalancer implementation has been around for a number of releases at this point and we are just extending it.
+
+#### Dev Preview -> Tech Preview
+
+NA
+
+#### Tech Preview -> GA
+
+NA
+
+#### Removing a deprecated feature
+
+NA
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades will be handled the same way they are for the current internal loadbalancer implementation. On upgrade the existing VIP configuration will be maintained. We will not automatically add additional VIPs to the cluster on upgrade. If an administrator of a dual-stack cluster wants to use the new functionality that will need to happen as a separate operation from the upgrade.
+
+When upgrading a cluster to the version that supports this functionality we will need to propagate new `PlatformSpec` fields from the current `PlatformStatus` fields. We have experience with this as when introducing dual-stack VIPs the [respective synchronization](https://github.com/openshift/cluster-network-operator/blob/2005bcd8c93de5bffc05c9c943b51386007f6b9a/pkg/controller/infrastructureconfig/sync_vips.go#L18-L22) has been implemented inside CNO.
+
+### Version Skew Strategy
+
+The keepalived configuration does not change between the version of the cluster and also when introducing this feature. There is no coordination needed and as mentioned above, enabling this functionality should be a separate operation from upgrading the cluster.
+
+### Operational Aspects of API Extensions
+
+NA
+
+#### Failure Modes
+
+NA
+
+#### Support Procedures
+
+NA
+
+## Implementation History
+
+NA
+
+## Alternatives
+
+The original enhancement for dual-stack VIPs introduced an idea of creating a separate instance of keepalived for the second IP stack. It was mentioned as something with simpler code changes but following this path would mean we can have two fundamentally different architectures of OpenShift clusters running in the field.
+
+Because we want clusters with dual-stack VIPs to not differ when they were installed like this from converted clusters, this is probably not the best idea.
```

Comment:
Indeed. And since it's already possible to deploy a second VIP as a second keepalived instance[0] we wouldn't need an entire new feature for that (although the existing mechanism has drawbacks because you need to populate some image references manually). That approach was found wanting by users which is why dual stack VIPs were properly integrated in the first place.

0: https://github.com/yboaron/kepalived-v6